lib/src/map/gestures/services/key_trigger_drag_rotate.dart

```diff
@@ -1,8 +1,11 @@
 part of 'base_services.dart';
 
 /// Service to handle the key-trigger and drag gesture to rotate the map. This
-/// is by default a CTRL +
-class KeyTriggerDragRotateGestureService extends BaseGestureService {
+/// is by default a CTRL +.
+///
+/// Can't extend from [_ProgressableGestureService] because of different
+/// method signatures.
```

Comment:
To avoid this, can we not use generic typing on the `_ProgressableGestureService`, and have the `start` argument be generically typed, and potentially named?
Then normally it can be `ScaleUpdateStart`, but in this case it can be `Null`? So the restriction on the generic type would be `ScaleUpdateStart?`?

Reply:
While I agree that it makes sense to have generics for variants of classes (e.g. Point<double> to have a double variant of Point), I don't think that introducing a generic for the type of a parameter type for only one function is good practice. Why try to force an interhitance on them if they **look** similar but **aren't** actually that similar?

Reply:
Ok, I understand that.

---

I see what you mean, but I would argue that `void submit(TapUpDetails _)` isn't ideal either. I'm struggling to understand why teritary tap is different to secondary tap or single tap? Why does it have a different signature? Am I just being thick?

Reply:
@JaffaKetchup The other gestures use the normal tap callback. Only the `onTertiaryTap` isn't provided by the `GestureDetector`.
We could use the `-up` callbacks for the tap, secondary-tap and tertiary-tap gestures. 

However the double-tap gesture does **not** have a `-up` callback. That's why we have to use the `onDoubleTapDown` callback.

My logic while implementing was to prefer using the normal tap callback and to use the -up callback if the other is not not available. if you want I can change this to whatever you like.

Reply:
discussed internally